### DISCUSSION TITLE
Creating plugin

### DISCUSSION THREAD

**Discussion by ivsaneesh** [vendurehq/vendure#1778] [Q&A]
How to create plugin. I was trying to create plugin using the details in document, but not able to run the plugin in the latest vendure version, The document is very old compared to the vendure latest version.

**Comment by michaelbromley** [comment]
Hi,
 
Do you mean these docs? https://www.vendure.io/docs/plugins/writing-a-vendure-plugin/

The docs should be up-to-date - the plugin API has not changed in the past year or so since v1.0.

What exact issues do you run into when trying to run your plugin?

**Reply by ivsaneesh** [reply]
In these document https://www.vendure.io/docs/plugins/writing-a-vendure-plugin/
Step 8 is not clear.  Where exactly to add an instance of plugin.

**Comment by ivsaneesh** [comment]
This is the error i am facing.

`import { Args, Mutation, Resolver } from '@nestjs/graphql';`

![vendurebug](https://user-images.githubusercontent.com/34297370/189849458-132548ab-da43-47d8-a5f7-31d8555d473d.png)

**Reply by michaelbromley** [reply]
ok. This is a typescript setting, so you have 3 options:

1. In your tsconfig.json file, set `noImplicitAny: false`
2. Add `args: any` explicitly
3. Add an actual correct type for args, i.e. `args: { id: string; }`

**Reply by ivsaneesh** [reply]
Thank you
